### PR TITLE
nixos/resilio: support secret files

### PR DESCRIPTION
**Pull request by jwoudenberg**
###### Motivation for this change

The Resilio nixos module supports declaratively configuring Resilio using the `sharedFolders` attribute. Secrets for shared folders can be passed using the `secret` attribute, with the downside that these secrets will end up in the nix store.

The goal of this change is to allow shared folder secrets to be specified without requiring they be put in the nix store, without breaking backwards compatibility with existing Resilio module configurations. It follows the approach outlined in the issue below, for upstream packages that do not provide password file support out of the box:
https://github.com/NixOS/nixpkgs/issues/24288

The change also modifies documentation to encourage the use of the new secret file support, because it seems like the safer default to me, but I don't feel strongly about that and am happy to change that back.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](./CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jwoudenberg** [comment]
Pinging @domenkozar, @thoughtpolice, and @cwoac as the owners of the resilio-sync package. Since this module doesn't have any maintainers listed (yet) I don't think you'd otherwise get a ping, but I might be wrong. Sorry for the spam!

**Comment by jwoudenberg** [comment]
I rebased this on master and added some release notes. I've also been running this code for a good couple of months now!

I'd love a review! Also, I'd be happy to help out as a maintainer of this package.

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 1]
```diff
@@ -160,14 +160,17 @@ In addition to numerous new and upgraded packages, this release has the followin
...
   ```nix
   {
     users.users.foo = {
       isSystemUser = true;
     };
   }
   ```
+
   also create a group for your user:
+
```
Comment (dali99) [inline nixos/doc/manual/release-notes/rl-2111.section.md:173]: These changes are unrelated to your PR and should not be included in its commit

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 2]
```diff
@@ -347,11 +350,11 @@ In addition to numerous new and upgraded packages, this release has the followin
...
 
   Sites definitions that use the old interface are automatically migrated in the new option. This backward compatibility will be removed in 22.05.
 
-- The dokuwiki module provides a new interface which allows to use different webservers with the new option [`services.dokuwiki.webserver`](options.html#opt-services.dokuwiki.webserver).  Currently `caddy` and `nginx` are supported. The definitions of dokuwiki sites should now be set in [`services.dokuwiki.sites`](options.html#opt-services.dokuwiki.sites).
+- The dokuwiki module provides a new interface which allows to use different webservers with the new option [`services.dokuwiki.webserver`](options.html#opt-services.dokuwiki.webserver). Currently `caddy` and `nginx` are supported. The definitions of dokuwiki sites should now be set in [`services.dokuwiki.sites`](options.html#opt-services.dokuwiki.sites).
```
Comment (dali99) [inline nixos/doc/manual/release-notes/rl-2111.section.md:357]: These changes are unrelated to your PR and should not be included in its commit

**Review by D4ndellion — CHANGES_REQUESTED** [review]
Diff looks otherwise pretty reasonable

Will test properly later

File: nixos/doc/manual/release-notes/rl-2111.section.md (hunk 1 above)
Comment (jwoudenberg) [reply nixos/doc/manual/release-notes/rl-2111.section.md:173]: Oops, good catch! Fixing it now.

**Comment by jwoudenberg** [comment]
Hey @dali99, thank you for your review comments! I addressed the problem you spotted, rebased these changes on master, and added myself as a maintainer of the resilio-sync package (would be happy to help review further PRs for these packages).

**Comment by jwoudenberg** [comment]
Hey @dali99, do you think you might have time to take another look at this? Totally fair if not of course!

**Comment by jwoudenberg** [comment]
I've been using a local version of this for a year now and it works great! I'd still love to get this merged, and as part of that to become a co-maintainer of this module so I can help review similar PRs in the future. If someone is up to review this, please let me know in reply and I'll dust this PR of!

**Comment by RyanGibb** [comment]
This would be very useful for myself as well! Thanks @jwoudenberg.

I notice @rnhmjoj and @cole-h, you've touched resolio recently. Do you think you could have a look at this by any chance?

**Review by thoughtpolice — APPROVED** [review]
lgtm 👍🏻

**Comment by aristidb** [comment]
It seems this PR broke the config for people not using sharedFolders config. I have a PR to fix that: #205633

And it seems do does @jcumming
